base_local_planner/src/obstacle_cost_function.cpp

```diff
@@ -89,7 +89,10 @@ double ObstacleCostFunction::scoreTrajectory(Trajectory &traj) {
     }
   }
 
-  for (unsigned int i = 0; i < traj.getPointsSize(); ++i) {
+  const unsigned int point_size = traj.getPointsSize();
```

Comment:
what is the reason for eliminating it. Yes it is not very useful but I don't see any issue in having the first point calculated.

Reply:
Primarily, the first point being in collision would invalidate all trajectories. If the robot can get out in one step (2.5cm granularity), meaning moving away from the obstacle, it should be allowed to do it.
Generally, it doesn't make sense to have the first point contribute to the cost, since it is the same for all trajectories. Currently the end point is used as "cost", but there is also a configuration for summing them up, and in the upstream branch the max is used. In the latter two cases, using the first point would just even out all the trajectories' costs, which is the opposite of what a cost function is supposed to do.